src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -166,6 +166,54 @@
         type="udf_body",
         trim_chars=("$",),
     ),
+    S3Path=RegexParser(
+        r"'s3://[a-z0-9][a-z0-9\.-]{1,61}[a-z0-9](?:/.+)?'",
+        CodeSegment,
+        name="s3_path",
+        type="bucket_path",
+    ),
+    GCSPath=RegexParser(
+        r"'gcs://[a-z0-9][\w\.-]{1,61}[a-z0-9](?:/.+)?'",
+        CodeSegment,
+        name="gcs_path",
+        type="bucket_path",
+    ),
+    AzureBlobStoragePath=RegexParser(
+        r"'azure://[a-z0-9]{3,24}\.blob\.core\.windows\.net/[a-z0-9][a-z0-9\.-]{1,61}[a-z0-9](?:/.+)?'",
+        CodeSegment,
+        name="azure_blob_storage_path",
+        type="bucket_path",
+    ),
```

Comment:
Yeah, I got them from the official docs and just added links to those as comments. That was worthwhile, not least because I realised I had the wrong account name for Azure Blob Storage accounts!